src/stan/io/deserializer.hpp

```diff
@@ -338,6 +341,29 @@ class deserializer {
     }
   }
 
+  /**
+   * Return an `std::vector`
+   * @tparam Ret The type to return.
+   * @tparam Sizes integral types.
+   * @param m The size of the vector.
+   * @param dims a possible set of inner container sizes passed to subsequent
+   * `read` functions.
+   */
+  template <typename Ret, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr,
+            require_same_t<value_type_t<Ret>, T>* = nullptr>
+  inline auto read(Eigen::Index m) {
```

Comment:
[optional] I doubt there is any benefit to separating out this case. Constructing `std::vector` from two iterators needs to make copies just as well.

Reply:
So the thing here is not actually about the construction, but just making sure we are only calling `check_r_capacity()` once instead of `m` times